src/graph/validator/MatchValidator.cpp

```diff
@@ -527,8 +525,8 @@ Status MatchValidator::validateUnwind(const UnwindClause *unwindClause,
 }
 
 StatusOr<Expression *> MatchValidator::makeSubFilter(const std::string &alias,
-                                                     const MapExpression *map,
-                                                     const std::string &label) const {
+                                                     MapExpression *map,
+                                                     const std::string &label) {
```

Comment:
Expressions related implementation changes should not change the Interface of the Validator.

Reply:
need to rewrite the expression after fold  into the mapexpression